src/NuGet.Clients/NuGet.PackageManagement.VisualStudio/Services/ExternalSettingsProviderService.cs

```diff
@@ -0,0 +1,229 @@
+        {
+            switch (moniker)
+            {
+                case MonikerAllowRestoreDownload:
+                    {
+                        if (value is bool boolValue)
+                        {
+                            PackageRestoreConsent.IsGrantedInSettings = boolValue;
+                            return Task.FromResult((ExternalSettingOperationResult)ExternalSettingOperationResult.Success.Instance);
+                        }
+                        break;
+                    }
+                case MonikerPackageRestoreAutomatic:
+                    {
+                        if (value is bool boolValue)
+                        {
+                            PackageRestoreConsent.IsAutomatic = boolValue;
+                            return Task.FromResult((ExternalSettingOperationResult)ExternalSettingOperationResult.Success.Instance);
+                        }
+                        break;
+                    }
+                case MonikerSkipBindingRedirects:
+                    {
+                        if (value is bool boolValue)
+                        {
+                            BindingRedirectBehavior.IsSkipped = boolValue;
+                            return Task.FromResult((ExternalSettingOperationResult)ExternalSettingOperationResult.Success.Instance);
+                        }
+                        break;
+                    }
+                case MonikerDefaultPackageManagementFormat:
+                    {
+                        if (value is string strValue)
+                        {
+                            PackageManagementFormat.SelectedPackageManagementFormat = strValue switch
+                            {
+                                MonikerPackagesConfig => 0,
+                                MonikerPackageReference => 1,
+                                _ => throw new ApplicationException("Error saving setting!"),
+                            };
+
+                            PackageManagementFormat.ApplyChanges();
+
+                            return Task.FromResult((ExternalSettingOperationResult)ExternalSettingOperationResult.Success.Instance);
+                        }
+                        break;
+                    }
+                case MonikerShowPackageManagementChooser:
+                    {
+                        if (value is bool boolValue)
+                        {
+                            PackageManagementFormat.Enabled = boolValue;
+                            PackageManagementFormat.ApplyChanges();
+                            return Task.FromResult((ExternalSettingOperationResult)ExternalSettingOperationResult.Success.Instance);
+                        }
+                        break;
+                    }
+                default: break;
+            }
+
+            throw new ApplicationException("Unknown setting!");
```

Comment:
just a thought, when the moniker is known, the code silently ignores when the value has the wrong type, so it doesn't feel entirely consistent to me to throw when the moniker is unknown.